src/ol_infrastructure/applications/mitxpro/__main__.py

```diff
@@ -0,0 +1,220 @@
+        ec2.SecurityGroupIngressArgs(
+            cidr_blocks=[mitxpro_vpc["cidr"]],
+            ipv6_cidr_blocks=[mitxpro_vpc["cidr_v6"]],
+            protocol="tcp",
+            from_port=18040,
+            to_port=18040,
+            description="Xqueue access",
+        ),
+    ],
+    tags=aws_config.tags,
+    vpc_id=mitxpro_vpc["id"],
+)
+
+mitxpro_edx_worker_security_group = ec2.SecurityGroup(
+    f"mitxpro-edx_worker-access-{env_suffix}",
+    name=f"mitxpro-edx_worker-access-{env_suffix}",
+    description="Access control to mitxpro_edx_worker",
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
+            security_groups=[mitxpro_edxapp_security_group.id],
+            protocol="tcp",
+            from_port=18040,  # noqa: WPS432
+            to_port=18040,  # noqa: WPS432
+            description="Xqueue access",
+        )
+    ],
+    tags=aws_config.tags,
+    vpc_id=mitxpro_vpc["id"],
+)
+
+mitxpro_edxapp_db_security_group = ec2.SecurityGroup(
+    f"mitxpro-edxapp-db-access-{env_suffix}",
+    name=f"mitxpro-edxapp-db-access-{env_suffix}",
+    description="Access from the mitxpro VPC to the mitxpro edxapp database",
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
+            security_groups=[
+                mitxpro_edxapp_security_group.id,
+                mitxpro_edx_worker_security_group.id,
+                consul_server_security_group.id,
+            ],
+            protocol="tcp",
+            from_port=3306,  # noqa: WPS432
+            to_port=3306,  # noqa: WPS432
+        )
+    ],
+    tags=aws_config.tags,
+    vpc_id=mitxpro_vpc["id"],
+)
+
+mitxpro_edxapp_db_config = OLMariaDBConfig(
+    instance_name=f"ol-mitxpro-edxapp-db-{env_suffix}",
+    password=get_config("mitxpro_edxapp:db_password"),
+    subnet_group_name=mitxpro_vpc["rds_subnet"],
+    security_groups=[mitxpro_edxapp_db_security_group],
+    tags=aws_config.tags,
+    db_name="edxapp_{db_purpose}".format(
+        db_purpose=(get_config("mitxpro_edxapp:db_purpose"))
+    ),
+    **defaults(stack)["rds"],
+)
+mitxpro_edxapp_db = OLAmazonDB(mitxpro_edxapp_db_config)
+
+mitxpro_edxapp_db_vault_backend_config = OLVaultMariaDatabaseConfig(
+    db_name=mitxpro_edxapp_db_config.db_name,
+    mount_point=f"{mitxpro_edxapp_db_config.engine}-mitxpro-edxapp-{mitxpro_environment}",
+    db_admin_username=mitxpro_edxapp_db_config.username,
+    db_admin_password=get_config("mitxpro_edxapp:db_password"),
+    db_host=mitxpro_edxapp_db.db_instance.address,
+)
+mitxpro_edxapp_db_vault_backend = OLVaultDatabaseBackend(
+    mitxpro_edxapp_db_vault_backend_config
+)
+
+mitxpro_edxapp_db_consul_node = Node(
+    "mysql", name="mysql", address=mitxpro_edxapp_db.db_instance.address
+)
+
+mitxpro_edxapp_db_consul_service = Service(
+    "mysql",
+    node="mysql",
+    name="mysql",
+    port=3306,
+    meta={
+        "external-node": True,
+        "external-probe": True,
+    },
+    checks=[
+        {
+            "check_id": "mitxpro_edxapp_db",
```

Comment:
It would be good to pass this as an instance of `ServiceCheckArgs` to allow for type enforcement.